torch/quantization/fx/fuse.py

```diff
@@ -0,0 +1,155 @@
+        node = node.args[0]
+        assert node.op == 'call_module'
+        assert isinstance(quantizer.modules[node.target], torch.nn.modules.Linear)
+        self.linear_node = node
+        self.linear = quantizer.modules[self.linear_node.target]
+
+    def fuse(self, quantizer, load_arg):
+        # since relu can be used multiple times, we'll need to create a relu module for each match
+        if self.relu_node.op == 'call_module':
+            relu = torch.nn.ReLU(quantizer.modules[self.relu_node.target].inplace)
+        else:
+            # TODO: get inplace argument from functional
+            relu = torch.nn.ReLU()
+        relu.training = self.linear.training
+        # linear_relu
+        linear_parent_name, linear_name = _parent_name(self.linear_node.target)
+        setattr(quantizer.modules[linear_parent_name], linear_name, torch.nn.intrinsic.LinearReLU(self.linear, relu))
+        return quantizer.fused_graph.node_copy(self.linear_node, load_arg)
+
+class Fuser:
+    def fuse_conv_bn(self, model, inplace=False):
+        input_root = model.root
+        if not inplace:
+            input_root = copy.deepcopy(input_root)
+        input_graph = model.graph
+        self.modules = dict(input_root.named_modules())
+
+        fusion_patterns = get_fusion_patterns()
+        # find conv-bn pairs
+        conv_bn_pairs = self._find_matches(input_root, input_graph, fusion_patterns)
+        self.fused_graph = Graph(delegate=DefaultDelegate())
+        env = {}
+
+        def load_arg(a):
+            return map_arg(a, lambda node: env[node.name])
+
+        for node in input_graph.nodes:
+            root_node, obj = conv_bn_pairs.get(node.name, (None, None))
+            if root_node is node:
+                env[node.name] = obj.fuse(self, load_arg)
+            elif root_node is None:
+                env[node.name] = self.fused_graph.node_copy(node, load_arg)
+            # node matched in patterns and is not root is removed here
+
+        self.fused_graph.output(load_arg(input_graph.result))
+        return GraphModule(input_root, self.fused_graph)
+
+    def _find_matches(self, root, graph, patterns):
+        modules = dict(root.named_modules())
+        match_map = {}  # node name -> (root_node, match_value?)
+
+        def apply_match(pattern, node, match):
+            if isinstance(pattern, tuple):
+                s, *args = pattern
+                apply_match(s, node, match)
+                for subpattern, arg in zip(args, node.args):
+                    apply_match(subpattern, arg, match)
+            else:
+                match_map[node.name] = match
+
+        for node in reversed(graph.nodes):
+            if node.name not in match_map:
+                for pattern, value in patterns.items():
+                    if matches(modules, node, pattern):
+                        apply_match(pattern, node, (node, value(self, node)))
+
+        return match_map
+
+def fuse(graph_module, inplace=False):
```

Comment:
Where do we call the `fuse()` function?

Reply:
before quantization, there is an example in test/quantization/test_quantize_fx.py